### PR TITLE
Mpa Grad Program & Grad Program Front-end: Req Bar

### PR DESCRIPTION
**Pull request by einc**
### Summary 

Added the MPA grad program. 
@willespencer integrated the grad program with his front-end changes! 

### Test Plan 
Try adding the MPA grad program. Make sure that at least one undergrad or grad program is selected! 

NEW front-end test plan
1. Try adding just a grad program, and make sure it displays on the requirements bar correctly, and has the color that colleges usually have
2. Try adding a college, 2 minors, and a grad program (but no major) and ensure everything shows correctly on the requirements bar, and that 3 different colors are used
3. Try adding a major back and ensure the grad program has the same color as the college, but other colors are unique.
4. Try any other permutations!

### Notes 

- The concentration requirement for this program is self-check. I was trying to use toggleable requirements but because all of the concentrations are suggestions, I didn't think toggleable reqs really worked so I kept it self-check for now.
- Some of the grad requirements are Yes/No requirements (similar to the Swim Test) so we should add those next sem
- The code to determine whether headers should display is quite complicated, and uses `reqIndex` and `numOfColleges` quite frequently. Left it mostly in-tact but added more to make sure grad programs work, but a refactor would make this code more simple and easy to change in the future.

### CODE REVIEW

File: src/requirements/data/grad/mpa.ts [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+      'Economic Analysis and Public Sector Economics 2',
+      'Quantitative Methods and Analytics 1',
+      'Quantitative Methods and Analytics 2',
+    ],
+    minNumberOfSlots: 6,
+  },
+  {
+    name: 'Additional Foundation Coursework',
+    description:
+      'Three additional semester-length courses (or equivalent) in any of the three Foundation Areas',
```
Comment (tcho6319) [inline src/requirements/data/grad/mpa.ts:33]: Nit: period

File: src/requirements/data/grad/mpa.ts [hunk 2]
```diff
@@ -0,0 +1,562 @@
...
+    name: 'Concentration Coursework',
+    description: 'Five courses within one of the eight concentration areas offered in the program.',
+    source: 'https://www.human.cornell.edu/cipa/academics/curriculum/guide',
+    fulfilledBy: 'self-check',
+    minCount: 5,
+  },
+  {
+    name: 'Professional Development Coursework',
+    description:
+      "Two courses to strengthen professional preparation, which may include statistical, analytical or mathematical skills, professional writing or speaking, leadership or management, or additional skills or knowledge needed for a fellow's selected concentration. CIPA Fellows may also want to consider including a course from the list of General Concentration courses, or from the list of Public and Nonprofit Management courses as one of their two specialized/professional development courses",
```
Comment (tcho6319) [inline src/requirements/data/grad/mpa.ts:158]: Nit: missing period at the end of the last sentence

File: src/requirements/data/grad/mpa.ts [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+      'PADM 5130',
+    ]),
+    fulfilledBy: 'courses',
+    perSlotMinCount: [2],
+    slotNames: ['Course'],
+  },
+  // TODO: PADM 5012 needs to be taken twice
+  {
+    name: 'Colloquium Coursework',
+    description: 'PADM 5009, PADM 5011, and PADM 5012 (must be taken twice)',
```
Comment (tcho6319) [inline src/requirements/data/grad/mpa.ts:553]: Period

File: src/utilities.ts [hunk 4]
```diff
@@ -51,18 +51,9 @@ export function getMinorFullName(acronym: string): string {
...
-    TEMP3: 'TEMP PROGRAM 3',
-  };
-
-  return acronym ? acronymDict[acronym] : '';
-
   // Return empty string if grad is not in requirementJSON
-  // const grad = acronym ? requirementJSON.grad[acronym] : null;
-  // return grad ? grad.name : '';
+  const grad = acronym ? requirementJSON.grad[acronym] : null;
+  return grad ? grad.name : '';
```
Comment (tcho6319) [inline src/utilities.ts:56]: The name of the grad program doesn't seem to show on req bar:
Screen Shot 2021-05-17 at 9 56 30 PM

@einc is this intended behavior or do we want to show the name of the grad program?
In the case that we do want to show the name, I think you'll have to modify RequirementHeader.vue, @willespencer.

**Review by tcho6319 — CHANGES_REQUESTED** [review]
Except for a few nits and a question about displaying the grad program name, this LGTM! @willespencer, I think Ein got to all the TODOs except the color one. Do you want to address that in her PR or wait for her to merge this one in before you make a new one?

Once my comments/questions are addressed, this PR should be g2g!

**Comment by einc** [comment]
> Except for a few nits and a question about displaying the grad program name, this LGTM! @willespencer, I think Ein got to all the TODOs except the color one. Do you want to address that in her PR or wait for her to merge this one in before you make a new one?
> 
> Once my comments/questions are addressed, this PR should be g2g!

@willespencer did the TODOs on my laptop LOL so i could test the grad program <3 I added the colors one to Notion (for next sem) bc its a small, nice to have feature but not important functionality wise!

File: cypress/integration/test.spec.ts [hunk 5]
```diff
@@ -170,10 +170,13 @@ it('Add a course with the new add modal (CS 1110)', () => {
   cy.get('[data-cyId=newCourse-requirementsDropdown]').click();
 
   // confirm we have the correct options by matching each dropdown element
-  let reqOptions = ['Advisor-Approved Electives', 'Major-approved Elective(s)', 'None'];
-  cy.get('[data-cyId=newCourse-reqOption]').each(($el, i) => {
-    cy.wrap($el).contains(reqOptions[i]);
-  });
+
+  // TODO - breaking only on the CI, investigate
```
Comment (tcho6319) [inline cypress/integration/test.spec.ts:174]: Can you make a Notion item for this if you haven't already?

**Review by tcho6319 — APPROVED** [review]
LGTM! Test plan works. Thanks Will and Ein :D 

> The code to determine whether headers should display is quite complicated, and uses reqIndex and numOfColleges quite frequently. Left it mostly in-tact but added more to make sure grad programs work, but a refactor would make this code more simple and easy to change in the future.

@willespencer can you make a Notion item?

**Pull request merged**